### ISSUE TITLE
Warn about `dict.fromkeys(keys, mutable)`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#4613]
`ruff` should error and suggest the following fix when passing a mutable 2nd arg to `dict.fromkeys()` since modifications will affect all keys which is a common pitfall.

```py
# bad
dict.fromkeys((1, 2), {})
dict.fromkeys((1, 2), [])
dict.fromkeys((1, 2), set())
# ...

# good
{key: {} for key in (1, 2)}
{key: [] for key in (1, 2)}
{key: set() for key in (1, 2)}
```

Examples:
- [StackoverFlow](https://stackoverflow.com/q/11509721)
- [SO 2](https://stackoverflow.com/q/8174723)
- [commit](https://github.com/CederGroupHub/chgnet/commit/c8ba05443280c280f904a7076f4635d3326bac66#diff-a7bc3c3838885a61fa9d82f50027518237bf263cbbcd33781f1176b7031f1275R184-R186)

**Comment by charliermarsh** [comment]
Oh interesting, I've never seen that!

**Comment by Skylion007** [comment]
Flake8-bugbear might be interested in adding this rule too, you should open an issue there as well.

**Comment by charliermarsh** [comment]
Marking as accepted, makes sense to me.

**Comment by tjkuson** [comment]
I'll implement this tomorrow (as a precursor to #9592)!
